### PR TITLE
Feature/limit order

### PR DESCRIPTION
**Pull request by yippee-ki-yay**

### CODE REVIEW

File: contracts/actions/exchange/LimitSellL2.sol [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+    /// @inheritdoc ActionBase
+    function actionType() public pure virtual override returns (uint8) {
+        return uint8(ActionType.STANDARD_ACTION);
+    }
+
+    //////////////////////////// ACTION LOGIC ////////////////////////////
+
+    /// @notice Sells a specified srcAmount for the dest token
+    /// @param _exchangeData DFS Exchange data struct
+    /// @param _from Address from which we"ll pull the srcTokens
```
Comment (majkic99) [inline contracts/actions/exchange/LimitSellL2.sol:82]: double quotes in we'll

File: contracts/actions/exchange/LimitSellL2.sol [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+    /// @param _to Address where we"ll send the _to token
+    /// @param _gasUsed Gas used for this strategy so we can take the fee
```
Comment (majkic99) [inline contracts/actions/exchange/LimitSellL2.sol:84]: add _l1GasUsed param

**Review by majkic99 — APPROVED** [review]
lgtm, some small natspec changes for LimitSellL2 needed, but looks ok, tests are passing, testing on forks looks well too

**Comment by majkic99** [comment]
But add new addresses to .json files

File: contracts/actions/exchange/LimitSell.sol [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+
+        // Reset the current price for the next strategy
+        tempStorage.setBytes32("CURR_PRICE", bytes32(0));
+        _exchangeData.dfsFeeDivider = 0;
+
+        if (_exchangeData.minPrice != currPrice) {
+            revert WrongPriceFromTrigger(currPrice, _exchangeData.minPrice);
+        }
+     
+        _exchangeData.srcAddr.pullTokensIfNeeded(_from, _exchangeData.srcAmount);
```
Comment (majkic99) [inline contracts/actions/exchange/LimitSell.sol:102]: if we write 
_exchangeData.srcAmount = _exchangeData.srcAddr.pullTokensIfNeeded(_from, _exchangeData.srcAmount);
we can delete lines 86-88

File: contracts/actions/exchange/LimitSellL2.sol [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+
+        // Reset the current price for the next strategy
+        tempStorage.setBytes32("CURR_PRICE", bytes32(0));
+        _exchangeData.dfsFeeDivider = 0;
+
+        if (_exchangeData.minPrice != currPrice) {
+            revert WrongPriceFromTrigger(currPrice, _exchangeData.minPrice);
+        }
+
+        _exchangeData.srcAddr.pullTokensIfNeeded(_from, _exchangeData.srcAmount);
```
Comment (majkic99) [inline contracts/actions/exchange/LimitSellL2.sol:110]: same thing here

**Review by majkic99 — APPROVED** [review]

File: contracts/actions/exchange/LimitSell.sol (hunk 3 above)
Comment (yippee-ki-yay) [reply contracts/actions/exchange/LimitSell.sol:102]: Good point, as the change is not critical and already deployed, will leave it as is for now.

**Pull request merged**